source/dxgi/dxgi_swapchain.cpp

```diff
@@ -313,6 +337,30 @@ HRESULT STDMETHODCALLTYPE DXGISwapChain::Present(UINT SyncInterval, UINT Flags)
 	const HRESULT hr = _orig->Present(SyncInterval, Flags);
 	g_in_dxgi_runtime = false;
 
+#if RESHADE_ADDON
+	// Emit PRESENT_END latency marker right after present returns
+	if (_is_initialized)
+	{
+		const uint64_t ts_ns = std::chrono::duration_cast<std::chrono::nanoseconds>(std::chrono::high_resolution_clock::now().time_since_epoch()).count();
```

Comment:
Add-ons can query a timestamp in the event callback(s), not much of a benefit to ReShade doing this.